cmd/sebak/cmd/run.go

```diff
@@ -164,11 +164,22 @@ func parseFlagsNode() {
 		flagEndpointString = nodeEndpoint.String()
 	}
 
-	if _, err = os.Stat(flagTLSCertFile); os.IsNotExist(err) {
-		common.PrintFlagsError(nodeCmd, "--tls-cert", err)
+	if len(flagTLSCertFile) > 0 {
+		if _, err = os.Stat(flagTLSCertFile); os.IsNotExist(err) {
+			common.PrintFlagsError(nodeCmd, "--tls-cert", err)
+		}
+	}
+	if len(flagTLSKeyFile) > 0 {
+		if _, err = os.Stat(flagTLSKeyFile); os.IsNotExist(err) {
+			common.PrintFlagsError(nodeCmd, "--tls-key", err)
+		}
 	}
-	if _, err = os.Stat(flagTLSKeyFile); os.IsNotExist(err) {
-		common.PrintFlagsError(nodeCmd, "--tls-key", err)
+	if strings.ToLower(nodeEndpoint.Scheme) == "https" && (len(flagTLSCertFile) < 1 || len(flagTLSKeyFile) < 1) {
```

Comment:
I think that whole block of code should be wrapped in the `https` check.
If the node does not expose an `https` endpoint, even if someone provides the flags, we ignore it.

Side question: Can an HTTPS node communicate with an HTTP node ?

Reply:
@spikeekips : I was suggesting this:
```go
if strings.ToLower(nodeEndpoint.Scheme) == "https" {
	 if len(flagTLSCertFile) < 1 || len(flagTLSKeyFile) < 1 {
		common.PrintFlagsError(
			nodeCmd,
			"--tls-cert, --tls-key",
			errors.New("HTTPS needs `--tls-cert` and `--tls-key`"),
		)

	if _, err = os.Stat(flagTLSCertFile); os.IsNotExist(err) {
		common.PrintFlagsError(nodeCmd, "--tls-cert", err)
	}

	if _, err = os.Stat(flagTLSKeyFile); os.IsNotExist(err) {
		common.PrintFlagsError(nodeCmd, "--tls-key", err)
	}
}
```

The point is, if we have the flags provided but we don't use https, just ignore them.